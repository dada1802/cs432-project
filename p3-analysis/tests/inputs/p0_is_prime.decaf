def bool is_prime(int n)
{
    int i;
    i = 2;

    if (n == 0 || n == 1)
    {
        return false;
    }

    while (i < n)
    {
        if (n % i == 0)
        {
            return false;
        }

        i = i + 1;
    }

    return true;
}

def int main()
{
    bool a;
    a = is_prime(4);
    return 0;
}