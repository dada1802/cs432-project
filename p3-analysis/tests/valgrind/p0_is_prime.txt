==3190261== Memcheck, a memory error detector
==3190261== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3190261== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3190261== Command: ../decaf inputs/p0_is_prime.decaf
==3190261== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 is_prime : (int) -> bool
 main : () -> int

  FuncDecl name="is_prime" return_type=bool parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:
     i : int

        Block [line 7]
        SYM TABLE:

        Block [line 12]
        SYM TABLE:

            Block [line 14]
            SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 24]
  SYM TABLE:

    Block [line 25]
    SYM TABLE:
     a : bool

==3190261== 
==3190261== HEAP SUMMARY:
==3190261==     in use at exit: 0 bytes in 0 blocks
==3190261==   total heap usage: 2,271 allocs, 2,271 frees, 294,408 bytes allocated
==3190261== 
==3190261== All heap blocks were freed -- no leaks are possible
==3190261== 
==3190261== For lists of detected and suppressed errors, rerun with: -s
==3190261== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
