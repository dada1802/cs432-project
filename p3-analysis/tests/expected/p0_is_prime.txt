Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 is_prime : (int) -> bool
 main : () -> int

  FuncDecl name="is_prime" return_type=bool parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:
     i : int

        Block [line 7]
        SYM TABLE:

        Block [line 12]
        SYM TABLE:

            Block [line 14]
            SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 24]
  SYM TABLE:

    Block [line 25]
    SYM TABLE:
     a : bool

